### ISSUE TITLE
Login HTTP requests without cookies with Form-based Authentication

### ISSUE THREAD

**Issue by zapbot** [zaproxy/zaproxy#1370]
```
Hi,

I've seen some web apps that always send a session cookie to its client even if before
logging in, and need this session cookie when users log in.
On the other hand, ZAP's Forced User function with Form-based Authentication seems
to send a HTTP request without cookies[1], so users can't login automatically with
ForcedUser Mode on.

* This kind of web apps have an Anti-CSRF token in its login form and the token value
seems to be stored in their session, so if you login without your session cookie, the
app cannot get the original token value and compare the one sent by a request with
it.

[1] https://code.google.com/p/zaproxy/source/browse/branches/2.3/src/org/zaproxy/zap/authentication/FormBasedAuthenticationMethodType.java#226
```

Original issue reported on code.google.com by `yukisov` on 2014-10-16 05:46:18

**Comment by KajanM** [comment]
> and need this session cookie when users log in

getting a fresh acsrf token and session cookie is addressed in #4682 
do we need to make any additional changes to fix this @thc202 ?

**Comment by KajanM** [comment]
:wave:

**Comment by thc202** [comment]
This was discussed in IRC and the decision was to send/get a session even if not in the presence of anti-csrf tokens. For this issue we just need to allow the user to specify the login page and access it prior sending the login request (this would also make the anti-csrf set up more straightforward).

**Comment by KajanM** [comment]
That makes sense.
So, `prepareRequestMessage` of `PostBasedAuthenticationMethod` needs to make a `GET` to the login page with the user set and no need to make  `GET` to login page in `replaceAntiCsrfTokenValueIfRequired`, right?

**Comment by thc202** [comment]
Right, the GET in `replaceAntiCsrfTokenValueIfRequired` would no longer be needed.

**Comment by thc202** [comment]
Fixed in #4696.
